src/SSHDebugPS/VsOutputWindowWrapper.cs

```diff
@@ -0,0 +1,138 @@
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    shell = Package.GetGlobalService(typeof(SVsUIShell)) as IVsUIShell;
+                }
+                catch (Exception)
+                {
+                }
+                return shell;
+            });
+        });
+
+
+        private class PaneInfo
+        {
+            public PaneInfo(Guid paneId)
+            {
+                this.paneId = paneId;
+                this.Shown = false;
+            }
+
+            internal Guid paneId;
+            internal bool Shown { get; set; }
+        }
+
+        private static Dictionary<string, PaneInfo> panes = new Dictionary<string, PaneInfo>()
+        {
+            // The 'Debug' pane exists by default
+            { "Debug", new PaneInfo(VSConstants.GUID_OutWindowDebugPane) }
+        };
+
+        /// <summary>
+        /// Writes text directly to the VS Output window.
+        /// </summary>
+        public static void Write(string message, string pane = "Debug")
+        {
+            ThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                try
+                {
+                    // Get the Output window
+                    IVsOutputWindow outputWindow = outputWindowLazy.Value;
+                    if (outputWindow == null)
+                    {
+                        return;
+                    }
+
+                    // Get the pane guid
+                    PaneInfo paneInfo;
+                    if (!panes.TryGetValue(pane, out paneInfo))
+                    {
+                        // Pane didn't exist, create it
+                        paneInfo = new PaneInfo(Guid.NewGuid());
+                        panes.Add(pane, paneInfo);
+                    }
+
+                    // Get the pane
+                    IVsOutputWindowPane outputPane;
+                    if (outputWindow.GetPane(ref paneInfo.paneId, out outputPane) != VSConstants.S_OK)
+                    {
+                        // Failed to get the pane - might need to create it first
+                        outputWindow.CreatePane(ref paneInfo.paneId, pane, fInitVisible: 1, fClearWithSolution: 1);
+                        outputWindow.GetPane(ref paneInfo.paneId, out outputPane);
+                    }
+
+                    // The first time we output text to a pane, ensure it's visible
+                    if (!paneInfo.Shown)
+                    {
+                        paneInfo.Shown = true;
+
+                        // Switch to the pane of the Output window
+                        outputPane.Activate();
+
+                        // Show the output window
+                        IVsUIShell shell = shellLazy.Value;
+                        if (shell != null)
+                        {
+                            Guid commandSet = VSConstants.GUID_VSStandardCommandSet97;
```

Comment:
why do you create a variable here for comandset, but then inline the OutputWindow CmdId?